### DISCUSSION TITLE
Call to undefined method Illuminate\Foundation\Auth\User::authentications()

### DISCUSSION THREAD

**Discussion by codebyray** [spatie/laravel-welcome-notification#43] [Q&A]
Everything works as expected when creating a new user, until the user sets their password. After the password has been set I receive the following error:

```php
[Call to undefined method Illuminate\Foundation\Auth\User::authentications()](https://dealerlogs.test/welcome/29?expires=1667893109&signature=944d9827a2f00acb40fa218fc3c8d9fda3fe0567e5e59474c122d2bf344158a6#top)
```
I am using:
PHP: 8.1.8
Laravel: 9.38.0

Any help with this would ne greatly appreciated.

**Comment by codebyray** [comment]
So @freekmurze , since this was moved to a discussion, it is not an issue?

It only happens when using this package and no customizations to the package or welcome emails.

Do you have the solution to the non-issue?

**Reply by freekmurze** [reply]
According to our tests, everything should be working as expected.

If you find an error in our package, please submit a PR to fix it.

**Reply by codebyray** [reply]
Thanks for your help, I'll move on and write my own, thought this would save me some time.

**Reply by nessimabadi** [reply]
This happens when using the rappasoft/laravel-authentication-log package.
Happens because the User model for the savePassword function is not the User model but the Authenticable one.

A solution is to create a new method in your WelcomeController that copies the code from the savePassword method that passes the User model and change the post route to point to the new method

**Comment by CodeEaster** [comment]
I am also facing the same issue, I have used ULID as the primary key for user ID and I am not sure if that is causing the issue.

After the Set initial password submission, I am getting that error.
![Screenshot 2023-09-30 at 3 38 36 PM](https://github.com/spatie/laravel-welcome-notification/assets/145494229/91486a47-1a07-4c02-9db5-bf3207c2f4f9)


Php Version : 8.1.23
Laravel Version: 10.25.2
